File: api/src/test/java/net/blueshell/api/factory/model/BoardDocumentFactory.java
package net.blueshell.api.factory.model;

import net.blueshell.api.testutil.ModelTestUtils;
import com.github.javafaker.Faker;
import lombok.RequiredArgsConstructor;
import net.blueshell.api.model.board.BoardDocument;
import org.springframework.stereotype.Component;

import java.util.concurrent.atomic.AtomicLong;

/**
 * Factory for BoardDocument model test instances.
 */
@Component
@RequiredArgsConstructor
public class BoardDocumentFactory {

    private static final AtomicLong counter = new AtomicLong(1000);
    private final Faker faker;
    private final BoardFactory boardFactory;
    private final FileFactory fileFactory;

    public BoardDocument createBasic() {
        BoardDocument bd = new BoardDocument();
        ModelTestUtils.setId(bd, generateId());
        bd.setBoard(boardFactory.createBasic());
        bd.setName(faker.book().title() + ".pdf");
        bd.setFile(fileFactory.createDocument());
        return bd;
    }

    public BoardDocument createFull() {
        return createBasic();
    }

    public BoardDocument createWithCustomizations(java.util.function.Consumer<BoardDocument> customizer) {
        BoardDocument bd = createFull();
        customizer.accept(bd);
        return bd;
    }

    private Long generateId() {
        return counter.incrementAndGet();
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/model/board/BoardDocumentFactory.kt
package net.blueshell.api.factory.model.board

import com.github.javafaker.Faker
import net.blueshell.api.board.persistence.BoardDocument
import net.blueshell.api.factory.model.FileFactory
import org.springframework.stereotype.Component
import java.util.concurrent.atomic.AtomicLong
import java.util.function.Consumer

/**
 * Factory for BoardDocument model test instances.
 */
@Component
class BoardDocumentFactory(
    private val faker: Faker,
    private val boardFactory: BoardFactory,
    private val fileFactory: FileFactory
) {

    fun createBasic(): BoardDocument {
        val boardDocument = BoardDocument()
        boardDocument.board = boardFactory.createBasic()
        boardDocument.name = faker.book().title() + ".pdf"
        boardDocument.file = fileFactory.createDocument()
        return boardDocument
    }

    fun createFull(): BoardDocument = createBasic()

    fun createWithCustomizations(customizer: Consumer<BoardDocument>): BoardDocument {
        val boardDocument = createFull()
        customizer.accept(boardDocument)
        return boardDocument
    }

    private fun generateId(): Long = COUNTER.incrementAndGet()

    private companion object {
        val COUNTER = AtomicLong(1000)
    }
}
